components/locid/src/subtags/region.rs

```diff
@@ -168,3 +200,66 @@ impl From<Region> for TinyAsciiStr<3> {
         input.0
     }
 }
+
+// Safety checklist for ULE:
+//
+// 1. Must not include any uninitialized or padding bytes (true since transparent over a ULE).
+// 2. Must have an alignment of 1 byte (true since transparent over a ULE).
+// 3. ULE::validate_byte_slice() checks that the given byte slice represents a valid slice.
+// 4. ULE::validate_byte_slice() checks that the given byte slice has a valid length.
+// 5. All other methods must be left with their default impl.
+// 6. Byte equality is semantic equality.
+#[cfg(feature = "zerovec")]
+unsafe impl zerovec::ule::ULE for Region {
+    fn validate_byte_slice(bytes: &[u8]) -> Result<(), zerovec::ZeroVecError> {
+        let it = bytes.chunks_exact(core::mem::size_of::<Self>());
+        if !it.remainder().is_empty() {
+            return Err(zerovec::ZeroVecError::length::<Self>(bytes.len()));
+        }
+        for v in it {
+            // The following can be removed once `array_chunks` is stabilized.
+            let mut a = [0; core::mem::size_of::<Self>()];
+            a.copy_from_slice(v);
+            if Self::try_from_raw(&a).is_err() {
```

Comment:
suggestion (nb): if try_from_raw is only used during validation should we expose it as `validate()` instead? Though it's not actually any additional code in the binary so perhaps it doesn't matter.

Reply:
I prefer to stick with the convention of validation being a fallible function where the return value is ignored. We specifically don't have very many `validate` functions in ICU4X and I'd rather not add one unless it's needed for a specific reason like avoid `unsafe` code (like in ULE) or if the constructor is expensive (allocates memory).

Reply:
works for me!